packages/portalnetwork/src/subprotocols/state/statedb.ts

```diff
@@ -0,0 +1,381 @@
+import { Debugger } from 'debug'
+import { StateProtocol } from './state.js'
+import { Trie } from '@ethereumjs/trie'
+import {
+  AccountTrieProofType,
+  ContractStorageTrieProofType,
+  StateNetworkContentType,
+} from './types.js'
+import { UintBigintType, fromHexString, toHexString } from '@chainsafe/ssz'
+import { Account, MapDB, equalsBytes } from '@ethereumjs/util'
+import { decodeStateNetworkContentKey } from './util.js'
+
+type StateRoot = string
+type StorageRoot = string
+type CodeHash = string
+type TrieRoot = string
+type Address = string
+
+export class StateDB {
+  trieDB: MapDB<string, string>
```

Comment:
Why aren't we using `level` here?

Reply:
we will.  i just sketched it out like this to get it working